File: problems_1000-2000/1431-kidsWithCandies/python/solution.py
class Solution:
    def kidsWithCandies(self, candies: list[int], extraCandies: int) -> list[bool]:
        max_candies = max(candies)
        return [candy + extraCandies >= max_candies for candy in candies]


if __name__ == "__main__":
    candies = [2, 3, 5, 1, 3]
    extraCandies = 3
    print(Solution().kidsWithCandies(candies, extraCandies))

File: problems_1000-2000/1431-kidsWithCandies/java/Solution.java
import java.util.ArrayList;
import java.util.List;

public class Solution {

    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max_candies = 0;
        for (int i = 0; i < candies.length; i++) {
            if (candies[i] > max_candies) {
                max_candies = candies[i];
            }
        }

        List<Boolean> result = new ArrayList<>();
        for (int i = 0; i < candies.length; i++) {
            if (candies[i] + extraCandies >= max_candies) {
                result.add(true);
            } else {
                result.add(false);
            }
        }

        return result;
    } 

    public static void main(String[] args) {
        int[] candies = {2,3,5,1,3};
        int extraCandies = 3;
        Solution solution = new Solution();
        System.out.println(solution.kidsWithCandies(candies, extraCandies));
    }
}